python/docs/source/development/testing.rst

```diff
@@ -0,0 +1,61 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+===============
+Testing PySpark
+===============
+
+In order to run PySpark tests, you should build Spark itself first via Maven
+or SBT. For example,
+
+.. code-block:: bash
+
+    build/mvn -DskipTests clean package
+
+After that, the PySpark test cases can be run via using ``python/run-tests``. For example,
+
+.. code-block:: bash
+
+    python/run-tests --python-executable=python3
+
+Note that:
+
+* If you are running tests on Mac OS, you may set ``OBJC_DISABLE_INITIALIZE_FORK_SAFETY`` environment variable to ``YES``.
+* If you are using JDK 11, you should set ``-Dio.netty.tryReflectionSetAccessible=true`` for Arrow related features. See also `Downloading <https://spark.apache.org/docs/latest/#downloading>`_.
```

Comment:
Just curious is this specifically necessary for debugging? shouldn't be necessary in Spark 3 in general.

Reply:
Yeah, ideally we should let it work out of the box but actually it is required to set `tryReflectionSetAccessible` property to run with JDK 11 (see https://github.com/apache/spark/pull/26552). Otherwise, the Arrow related code paths fail.  See also ARROW-7223.

Reply:
Oh! but we set it in testing scripts by default so it is more required for debugging when we don't use our own testing script. Yes, I got the point now. Let me just remove this here.